str/trtools/merge_str_runner.py

```diff
@@ -33,42 +31,51 @@
 # input directory
 @click.option('--input-dir', help='gs://...')
 # input sample ID
-@click.argument('external-wgs-ids', nargs=-1)
+@click.argument('internal-wgs-ids', nargs=-1)
 @click.command()
 def main(
-    dataset, input_dir, external_wgs_ids: list[str]
+    dataset, input_dir, internal_wgs_ids: list[str]
 ):  # pylint: disable=missing-function-docstring
     # Initializing Batch
     b = get_batch()
 
-    external_id_to_cpg_id: dict[str, str] = SampleApi().get_sample_id_map_by_external(
-        dataset, list(external_wgs_ids)
-    )
-
+    # Initialise TRTools job to run mergeSTR
     trtools_job = b.new_job(name='mergeSTR')
     trtools_job.image(TRTOOLS_IMAGE)
     trtools_job.cpu(8)
-    trtools_job.cloudfuse(f'cpg-{dataset}-main', '/vcffuse')
+    # mount using cloudfuse for reading input files
+    trtools_job.cloudfuse(f'cpg-{dataset}-main-analysis', '/vcffuse')
+    trtools_job.declare_resource_group(
+        vcf_output={
+            'vcf': '{root}.vcf',
+            'vcf.gz': '{root}.vcf.gz',
+            'vcf.gz.tbi': '{root}.vcf.gz.tbi',
+        }
+    )
 
+    # read in input file paths
     vcffuse_path = []
-    for id in list(external_id_to_cpg_id.values()):
+    for id in list(internal_wgs_ids):
         vcf = os.path.join(input_dir, f'{id}_eh.reheader.vcf.gz')
         suffix = vcf.removeprefix('gs://').split('/', maxsplit=1)[1]
         vcffuse_path.append(f'/vcffuse/{suffix}')
     num_samples = len(vcffuse_path)
     vcffuse_path = ','.join(vcffuse_path)  # string format for input into mergeSTR
 
-    output_path_vcf = output_path(f'mergeSTR_{num_samples}_samples_eh.vcf.bgz')
-    output_path_vcf = output_path_vcf.removeprefix('gs://').split('/', maxsplit=1)[1]
-
     trtools_job.command(
         f"""
-    mergeSTR --vcfs {vcffuse_path} --out temp.vcf --vcftype eh
-    bgzip -c temp.vcf > /vcffuse/{output_path_vcf}
-    tabix /vcffuse/{output_path_vcf}
+    mergeSTR --vcfs {vcffuse_path} --out {trtools_job.vcf_output} --vcftype eh
+    bgzip -c {trtools_job.vcf_output}.vcf > {trtools_job.vcf_output['vcf.gz']}
```

Comment:
This works fine, so just a nit here. 

You've made a `vcf` entry in the resource group, so you can use `{trtools_job.vcf_output['vcf'}` here to be consistent. 

Alternatively you could drop `vcf` from the resource group, then below when you have `b.write_output(..)` you can simplify this to

```
b.write_output(trtools_job.vcf_output, output_path_name)
```

which would automatically copy out the `vcf.gz` and `vcf.gz.tbi`